libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -667,42 +694,25 @@ private JSONArray getDataFromRow(Cursor cursor) throws JSONException {
 		JSONArray row = new JSONArray();
 		int columnCount = cursor.getColumnCount();
 		for (int i=0; i<columnCount; i++) {
-			String raw = cursor.getString(i);
-
-            if (raw == null) {
-                row.put(raw);
-            } else {
-    			// Is this column holding a serialized json object?
-    			if (cursor.getColumnName(i).endsWith(SOUP_COL)) {
-    				row.put(new JSONObject(raw));
-    				// Note: we could end up returning a string if you aliased the column
-    			}
-    			else {
-    				// TODO Leverage cursor.getType once our min api is 11 or above
-    				// For now, we do our best to guess
-    				
-    				// Is it holding a integer ?
-    	    		try {
-    	    			Long n = Long.parseLong(raw);
-    	    			row.put(n);
-    	    			// Note: we could end up returning an integer for a string column if you have a string value that contains just an integer
-    	    		}
-    	    		// Is it holding a floating ?
-    	    		catch (NumberFormatException e) {
-    	    			try { 
-    		    			Double d = Double.parseDouble(raw);
-    		    			// No exception, let's get the value straight from the cursor
-    		    			// XXX Double.parseDouble(cursor.getString(i)) is sometimes different from cursor.getDouble(i) !!!
-    		    			d = cursor.getDouble(i);
-    		    			row.put(d);
-    		    			// Note: we could end up returning an integer for a string column if you have a string value that contains just an integer
-    	    			}
-    		    		// It must be holding a string then
-    	    			catch (NumberFormatException ne) {
-    		    			row.put(raw);
-    	    			}
-    	    		}
-    			}
+            int valueType = cursor.getType(i);
```

Comment:
This is a change we could have made a long while back - getType was not available before API 11. See TODO on line 681 in the old code.